ipc/cli/src/commands/deploy/mod.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: MIT
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use clap::{Args, ValueEnum};
+use ethers::types::Address;
+use fendermint_eth_deployer::{EthContractDeployer, SubnetCreationPrivilege};
+use fendermint_eth_hardhat::Hardhat;
```

Comment:
will it create some kind of cycler dependency in the future?

Reply:
Good question. I thought about it quite a bit and I wasn't sure where exactly to put it. Also didn't want to re-shuffle too much. I honestly think that moving forward we should have some crates that are outside of ipc and fendermint and can be used by both. WDYT?